### PR TITLE
Misc. changes based on the recent feedback from Smarter Balanced.

### PR DESCRIPTION
**Pull request by agorina**
Added a table to control application and schema relationship.

### CODE REVIEW

File: reporting/sql/V201702141487123176__misc_fixes.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+** Remove is_demo flag from the student table; the rdw is not supposed to receive demo records.
+** Remove exam location.
+** Make an exam's session not optional.
+**/
+USE reporting;
+
+CREATE TABLE application_schema_version (
+   major_version int UNIQUE NOT NULL
+);
+
```
Comment (agorina) [inline reporting/sql/V201702141487123176__misc_fixes.sql]: The idea is that the application will check this version to determine if it is compatible with the db schema.
@malaffoon, do you think we should have major and minor versions?

File: reporting/sql/V201702141487123176__misc_fixes.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+INSERT INTO application_schema_version (major_version) VALUES (0);
+
+ALTER TABLE student DROP COLUMN is_demo;
+
```
Comment (agorina) [inline reporting/sql/V201702141487123176__misc_fixes.sql]: RDW is not supposed to receive/accept demo records.

File: reporting/sql/V201702141487123176__misc_fixes.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ALTER TABLE iab_exam
+  DROP FOREIGN KEY fk__iab_exam__exam_location,
+  DROP COLUMN asmt_session_location_id,
+  MODIFY session_id varchar(128) NOT NULL;
+
```
Comment (agorina) [inline reporting/sql/V201702141487123176__misc_fixes.sql]: We discussed that it is not delivered in TRT format and is not important for this phase.

File: reporting/sql/V201702141487123176__misc_fixes.sql (hunk 1 above)
Comment (malaffoon) [inline reporting/sql/V201702141487123176__misc_fixes.sql]: I'm waffling a bit on this: should we just have a string in there similar to software versioning, or do we break it up into major/minor? If we break it up i think we should have both major and minor: major controls backward compatibility while minor allows forward compatibility. OTOH, if we only ever query the minor version for informational purposes, why bother adding it?

Comment (agorina) [inline reporting/sql/V201702141487123176__misc_fixes.sql]: You nicely repeated my chain of thoughts :) 
I guess I will keep just a major version of now.

**Pull request merged**